### PR TITLE
Refactor ps_facetedsearch

### PR DESCRIPTION
**Pull request by jocel1**
Complete refactoring.
Goal of the refactoring:
- try to simplify the logic
- adapter structure to allow the use of other search engine (default MySQL)
- improve performances (do not iterate anymore on all the products to construct the filterBlock)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/ps_facetedsearch/21)

### CODE REVIEW

**Comment by jocel1** [comment]
Still WIP

**Comment by eternoendless** [comment]
Some PHPDoc would be nice ;)

File: ps_facetedsearch.php [hunk 1]
```diff
@@ -47,7 +44,7 @@ public function __construct()
     {
         $this->name = 'ps_facetedsearch';
         $this->tab = 'front_office_features';
-        $this->version = '2.0.0';
+        $this->version = '3.0.0';
```
Comment (maximebiloe) [inline ps_facetedsearch.php]: No need to change that, the release manager will do it

File: ps_facetedsearch.php [hunk 2]
```diff
@@ -864,108 +896,103 @@ public static function indexProductPrices($id_product, $smart = true)
...
-                    $max_price[$currency['id_currency']] = $price;
-                }
-                if ($price == 0) {
-                    continue;
-                }
-                if (is_null($min_price[$currency['id_currency']]) || $price < $min_price[$currency['id_currency']]) {
-                    $min_price[$currency['id_currency']] = $price;
+                // Get price by currency & country, without reduction!
+                foreach ($currency_list as $currency) {
+                    $price = Product::priceCalculation($id_shop, (int)$id_product, null, $id_country, null, null,
```
Comment (maximebiloe) [inline ps_facetedsearch.php]: Can you be careful with the indent?

File: src/NativeModuleFacetedSearchBundle/Ps_FacetedsearchFilterBlock.php [hunk 3]
```diff
@@ -0,0 +1,1148 @@
...
+    /**
+     * @param int $nbProducts
+     * @param array $selectedFilters
+     *
+     * @return array
+     */
+    public function getFilterBlock(
+        $nbProducts,
+        $selectedFilters
+    ) {
```
Comment (maximebiloe) [inline src/NativeModuleFacetedSearchBundle/Ps_FacetedsearchFilterBlock.php]: Can you open the bracket on a new line, please?

**Comment by jocel1** [comment]
Thx @maximebiloe , I've just commit the required changes for PHP 5.3 compat

**Comment by Quetzacoalt91** [comment]
Tested with https://github.com/PrestaShop/PrestaShop/pull/8246, it removes the depreaction notices I've met with this module. :)

**Comment by mmendozadelsolar** [comment]
Hello,

This PR will be merge?

**Comment by jocel1** [comment]
@mmendozadelsolar: it's still WIP, I've a few remaining bugs to fix before :)

**Comment by kpodemski** [comment]
@jocel1 is this still WIP? can we expect it to be released some day? :)

**Comment by jocel1** [comment]
Hi Krystian!

Well I really hope so, I need to test it a bit more but it should be almost done! (feel free to test it as well ;))

**Comment by PierreRambaud** [comment]
Close for #52 

Thanks @jocel1

**Pull request closed without merging**